src/main/java/org/isf/medicalstockward/service/MedicalStockWardIoOperations.java

```diff
@@ -215,7 +217,13 @@ protected boolean updateStockWardQuantity(
             if(medicalWardTo != null) {
             	repository.updateInQuantity(Math.abs(qty), wardTo, medical, lot);
             } else {
-                repository.insertMedicalWard(wardTo, medical, Math.abs(qty), lot);
+				MedicalWard medicalWard = new MedicalWard();
+				medicalWard.setWard(movement.getWard());
+				medicalWard.setMedical(movement.getMedical());
+				medicalWard.setInQuantity((float)Math.abs(qty));
+				medicalWard.setOutQuantity(0.0f);
+				medicalWard.setLot(movement.getLot());
+                repository.save(medicalWard);
             }
             repository.updateOutQuantity(Math.abs(qty), ward, medical, lot);
```

Comment:
Remove this line, no need to update the quantity after the previous .save(medicalWard)